File: Stock_Maintenance_System_Ui/Stock_Maintenance_System_Ui/src/app/models/Product/ProductRequest.ts
export interface ProductRequest {
    productName: string;
    companyId: number;
    categoryId: number;
    productCategoryId?: number;
    description?: string;
    mrp: number;
    salesPrice: number;
    taxType?: string;
    totalQuantity: number;
    barCode?: string;
    brandName?: string;
    taxPercent?: number;
}

File: Stock_Maintenance_System_Api/ApiRequest/ProductRequest.cs
namespace InventorySystem_Api.ApiRequest;
public record ProductRequest(
    string ProductName,
    int CompanyId,
    int CategoryId,
    int? ProductCategoryId,
    string? Description,
    decimal Mrp,
    decimal SalesPrice,
    decimal LandingPrice,
    int TotalQuantity,
    bool IsActive = false
);